src/server/parser/parser.ts

```diff
@@ -1714,6 +1716,74 @@ export class Parser
 		return result
 	}
 
+	parseEnumDef(): Result<ASTNode, ParsingErrors>
+	{
+		const token = this.lexer.token.clone()
+		const match = this.match(TokenType.enumDef)
+		if (!match)
+			return Err('UnreachableState')
+
+		const ident = this.parseIdent()
+		if (!ident.isDefined())
+			return Err('InvalidTokenSequence')
+		if (ident.isErr() || ident.isInvalid())
+			return ident
+
+		const enumName = ident.val
+		// If the symbol's already in the table but is not a function symbol, that's an error
+		if (enumName.symbol && !enumName.symbol.type.mask(SymbolTypes.enum))
+			return Err('SymbolAlreadyDefined')
+		enumName.symbol = new MangroveSymbol(enumName.value, new SymbolType(SymbolTypes.struct | SymbolTypes.type))
+		this.symbolTable.insert(enumName.symbol)
+
+		const block = this.parseBlock({allowExtStmt: false, isEnum: true})
+		if (!block.isDefined())
+			return Err('InvalidTokenSequence')
+		if (block.isErr() || block.isInvalid())
+			return block
+		const astBlock = block.val as ASTBlock
+		const enumValues = astBlock.statements
+		const valuesValid = enumValues.every(value => value.valid)
+		if (enumValues && !valuesValid)
+			this.symbolTable.pop(this)
+		return Ok(new ASTEnum(token, enumName, enumValues as ASTEnumValue[]))
+	}
+
+	parseEnumValue(): Result<ASTEnumValue, ParsingErrors>
+	{
+		let value
+		if (this.lexer.token.typeIsOneOf(TokenType.comma))
```

Comment:
Given that `token` is used repeatedly, please pull this out as a `const` - the token object referenced does not actually change, only the contents of the token, which is why this pattern works in the rest of the parser

Reply:
Sorted!